fastlane/lib/fastlane/actions/xcode_select.rb

```diff
@@ -18,32 +18,92 @@ module Actions
     #   environment variable into the full Developer content path.
     #
     class XcodeSelectAction < Action
-      def self.run(params)
-        params = nil unless params.kind_of? Array
-        xcode_path = (params || []).first
 
-        # Verify that a param was passed in
-        UI.user_error!("Path to Xcode application required (e.x. \"/Applications/Xcode.app\")") unless xcode_path.to_s.length > 0
+      def self.find_xcode(req)
```

Comment:
I'm not a big fan with adding new methods here, could we move this to a helper class? Similar to `crashlytics_helper.rb`